rod_cutting_problem/rod_cutting.py

```diff
@@ -0,0 +1,37 @@
+"""
+    This module calculates optimized solution for rod cutting
+    by function rod_cutting() with arguments as defined
+    in main()
+"""
+
+
+def rod_cutting(price, rod_len):
+    """
+    Computes maximum money that can be earned by cutting
+    a rod of length rod_len (Bottom-Up Approach).
+
+    Time Complexity : O(rod_len^2)
+    Space Complexity : O(rod_len)
+    :param price: List in which price[i] denotes price of rod of length i.
+    :param rod_len: Total length of rod to be considered for optimal amount.
+    :return: returns optimal solution for rod of length rod_len.
+    """
+    opt_price = [0] * (rod_len + 1)
+
+    for i in range(1, rod_len + 1):
+        for j in range(i):
+            opt_price[i] = max(max(opt_price), price[j] + opt_price[i - j - 1])
```

Comment:
```python
for i in range(1, rod_len + 1):
   opt_price[i] = max([-1] + [price[j] + opt_price[i - j - 1] for j in range(i)])
```

There may be errors, but this is what I was trying to say. Please check thoroughly and push.